src/Session.php

```diff
@@ -116,7 +116,7 @@ public function readSession($id)
     {
         $data = $this->redis->executeCommand('GET', [$this->calculateKey($id)]);
 
-        return $data === false || $data === null ? '' : $data;
+        return $data === false || $data === null ? '' : strval($data);
```

Comment:
what was data in your case? false and null cases are already covered here. Are you sure you are using the latest verion of redis extension?

Reply:
"# Server
redis_version:4.0.2
redis_git_sha1:00000000
redis_git_dirty:0
redis_build_id:9338724bc3b9b1b3
redis_mode:standalone
os:Linux 3.10.0-693.21.1.el7.x86_64 x86_64
arch_bits:64
multiplexing_api:epoll
atomicvar_api:atomic-builtin
gcc_version:4.8.5
process_id:60
run_id:cf0d389d1eb2de42c664f1b6bda405986d3569b1
tcp_port:6379

Reply:
PHP Version 7.1.23
Yii :2.0.15

Reply:
which version of the redis extension?

`composer show |grep yii2-redis`

Reply:
2.0.4

Reply:
ok,i try